RevitClasher/Tools.cs

```diff
@@ -167,35 +163,68 @@ public static bool belongsToView(View3D view3D, BoundingBoxXYZ bb)
+            elementsList.Add(element.Id);
 
-            bool output = false;
+            bool clashOutputFlag = false;
 
             //Look for each solid in linked model
-            foreach (var geomObj in a)
+            foreach (GeometryObject geomObj in geometryElement)
             {
-                Solid geomSolid = geomObj as Solid;
-                if (null != geomSolid)
+                // If solid is present we perform a solid intersection filter.
+                // and later, perform a filter iteration with the element to clash (Element A vs Element B)
+                if(geomObj is Solid)
                 {
+                    Solid solidObject = geomObj as Solid;
+                    ElementIntersectsSolidFilter elementIntersectsSolidFilter = new ElementIntersectsSolidFilter(solidObject);
+                    FilteredElementCollector collector = new FilteredElementCollector(doc, elementsList).WherePasses(elementIntersectsSolidFilter);
+                    
+                    // If collector count is zero, this means that Solid intersection can't be possible with the given element (ie: Fabrication Parts).
+                    if(collector.Count() == 0)
+                    {
+                        BoundingBoxXYZ bbox = geometryElement.GetBoundingBox();
+                        Outline outline = new Outline(bbox.Min, bbox.Max);
+                        BoundingBoxIntersectsFilter bbFilter = new BoundingBoxIntersectsFilter(outline);
+                        collector = new FilteredElementCollector(doc, elementsList).WherePasses(bbFilter);
+                    }
 
-                    ElementIntersectsSolidFilter elementIntersectsSolidFilter = new ElementIntersectsSolidFilter(geomSolid);
-                    FilteredElementCollector collector = new FilteredElementCollector(doc, l).WherePasses(elementIntersectsSolidFilter);
                     foreach (var item in collector)
                     {
-                        output = true;
+                        clashOutputFlag = true;
                         break;
                     }
+                }
 
+                // If a Geometry instance is present. We implement a BoundingBox intersect filter over the Geometry of element A
+                // and later, perform a Filter iteration with the element to clash (Element A vs Element B)
+                if(geomObj is GeometryInstance)
+                {
+                    BoundingBoxXYZ bbox = geometryElement.GetBoundingBox();
+                    Outline outline = new Outline(bbox.Min, bbox.Max);
+                    BoundingBoxIntersectsFilter bbFilter = new BoundingBoxIntersectsFilter(outline);
+                    FilteredElementCollector collector = new FilteredElementCollector(doc, elementsList).WherePasses(bbFilter);
+                    foreach (var item in collector)
+                    {
+                        clashOutputFlag = true;
+                        break;
+                    }
```

Comment:
Some elements return a `GeometryInstance` instead of Solid, for those elements is necesary to apply a Bounding Box intersection.